pkg/api/api.go

```diff
@@ -48,62 +51,83 @@ func New(profileCatalog Catalog) API {
 // ProfilesHandler is the handler for /profiles requests.
 func (a *API) ProfilesHandler(w http.ResponseWriter, r *http.Request) {
 	query := r.URL.Query().Get("name")
+	logger := a.logger.WithValues("endpoint", "/profiles", "name", query)
```

Comment:
this makes all of the below logs pre-fixed with these key-values

Reply:
we could maybe use `r.URL.Path` instead of "/profiles" ?